org.idempiere.test/src/org/idempiere/test/base/DocumentStatusTest.java

```diff
@@ -0,0 +1,144 @@
+package org.idempiere.test.base;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+
+import org.compiere.model.MDocumentStatus;
+import org.compiere.model.MEntityType;
+import org.compiere.model.MInvoice;
+import org.compiere.model.MOrder;
+import org.compiere.model.MTable;
+import org.compiere.model.PO;
+import org.compiere.util.DB;
+import org.compiere.util.Env;
+import org.idempiere.test.AbstractTestCase;
+import org.junit.jupiter.api.Test;
+
+public class DocumentStatusTest extends AbstractTestCase {
+
+	public DocumentStatusTest() {
+	}
+
+	private static final int USER_GARDENADMIN = 101;
+	private static final int USER_GARDENUSER = 102;
+	private static final int USER_SUPERUSER = 100;
+
+	private static final int ROLE_GARDENWORLDADMIN = 102;
+	private static final int ROLE_GARDENWORLDUSER = 103;
+	private static final int ROLE_GARDENWORLD_ADMIN_NOT_ADVANCED = 200001;
+
+	private static final int WINDOW_ID_INVOICE_CUSTOMER = 167;
+	private static final int WINDOW_ID_SALES_ORDER = 143;
+	private static final String TABLENAME_PA_DOCUMENTSTATUSACCESS = "PA_DOCUMENTSTATUSACCESS";
+
+	/**
+	 * https://idempiere.atlassian.net/browse/IDEMPIERE-4836
+	 */
+	@Test
+	public void testDocumentStatusVisibility() {
+		int documentSeqID = DB.getNextID(Env.getAD_Client_ID(Env.getCtx()), MDocumentStatus.Table_Name, getTrxName());
+
+		/* Document Status with specific role and user access */
+		MDocumentStatus invoiceDS = (MDocumentStatus) MTable.get(Env.getCtx(), MDocumentStatus.Table_ID).getPO(0,
```

Comment:
this is not recommended, should just use the MDocumentStatus constructor instead.